### ISSUE TITLE
Could not find MIME for buffer (null)

### ISSUE THREAD

**Issue by Allaphy** [oliver-moran/jimp#643]
I have this problem "Could not find MIME for buffer (null)" and i don't know how to fix it.
I believe that the problem is in jimp.read.
that is the part of the code where i read the images:

**var imgs = [pa1, pa2];**   _// pa1 and pa2 are images URLs._
**var jimps = [];**
**for (var i = 0; i < imgs.length; i++) {
      jimps.push(await jimp.read(imgs[i]));
}**

'imgs[]' are the URLs of the images to read.

- Jimp Version: 0.5.3
- Operating System: Windows 8.1
- Node version: 8.11.4

**Comment by hipstersmoothie** [comment]
I’m probably going to need a link to one of your images. I’m pretty sure
image urls are working so I’ll need a test image

**Comment by DavidOndrus** [comment]
I experience it too but with files saved locally. I've been using old version (0.2.28) for a long time, then on 2nd October I made an update to 0.5.3. It never happened to me before. It's not reproducible when someone sends you a file because in my situation, for example only 2 of 5 tries crash. When it crashes and then I try the same file, after some time it works. It doesn't make sense.

Right now I tried the file I'm attaching, on 2nd try it worked.
![636629e3-dcee-4a34-9c07-99f7aadfbd99](https://user-images.githubusercontent.com/15279818/46790326-fcef2e80-cd3e-11e8-98c8-56e20c2275e3.jpg)

**Comment by Allaphy** [comment]
the urls aren't aways the same, are urls from discord's avatars (message.author.displayAvatarURL), for example: https://cdn.discordapp.com/avatars/472158887966015508/a64800726ff83aedd350092730484a1e.jpg?size=2048

with me, just some times crashes too, some times works fine, and some times it wont.

**Comment by hipstersmoothie** [comment]
The following runs without any errors. I'll do a run with a 1000 but I'm not sure this will reproduce it. 

Are you able to reproduce this outside the discord server?

```js
const Jimp = require('jimp');

test();

async function test() {
  for (let i = 0; i < 100; i++) {
    const image = await Jimp.read(
      'https://cdn.discordapp.com/avatars/472158887966015508/a64800726ff83aedd350092730484a1e.jpg?size=2048'
    );

    const buffer = image.getBufferAsync(Jimp.AUTO);
    console.log(i, buffer);
  }
}
```

**Comment by hipstersmoothie** [comment]
I feel like maybe the request is failing? you said it had to do with upload. maybe sometimes you do it too soon and the resource isnt quite there yet?

**Comment by DavidOndrus** [comment]
I'm able to reproduce it with local files, without network so it's not an issue with online files, it's a general Jimp.read issue.

**Comment by hipstersmoothie** [comment]
You you give me a script like the one i provided above? Currently 700 requests and no errors

**Comment by DavidOndrus** [comment]
Download my file and use your script with Jimp.read('/app/.tmp/uploads/636629e3-dcee-4a34-9c07-99f7aadfbd99.jpg')

**Comment by hipstersmoothie** [comment]
Still no luck. How often does it happen for you?

<img width="665" alt="screen shot 2018-10-11 at 11 35 57 pm" src="https://user-images.githubusercontent.com/1192452/46852048-72461800-cdae-11e8-8bce-7210ecdafba2.png">

**Comment by hipstersmoothie** [comment]
Can you put together a repo that demonstrates the issue? I so far have not been able to reproduce. I am also not on windows

**Comment by DavidOndrus** [comment]
I would not try to reproduce it with tests because even if one of many tries fails, you are not able to repeat the same process with the same result. I mean, if one line printed "null" in the test output above, it's not very useful information I think. I've searched this repo and found a line with the error string we've posted so I would go for investigating the stack trace. If similar errors occur, it's often a problem with thread (asynchronous operations) because it's reproducible e.g. only once per 10 tries.

**Comment by Allaphy** [comment]
send the buffer to cosole.log i know how to do it, but when i do this, it wont send the message on discord.

i must do something with this buffer ?

**Comment by Allaphy** [comment]
**var p1 = hguData[sort1].user; // String
var pa1 = hguData[sort1].avatar; // avatar URL (image)
var p2 = hguData[sort2].user; // String
var pa2 = hguData[sort2].avatar; // avatar URL (image)
var data;
var imgs = [pa1, pa2];
var jimps = [];

for (var i = 0; i < imgs.length; i++) {
	jimps.push(await jimp.read(imgs[i]));
}

Promise.all(jimps).then(function() {
	return Promise.all(jimps);
}).then(function(d) {
	data = d;
	return jimp.loadFont(jimp.FONT_SANS_16_WHITE);
}).then(async function(font) {
	msg.react("⏳");
	data[0].resize(75, 75);
	data[1].resize(75, 75);

	let nova = new jimp(255, 125, '#000000', (err, image) => {});
	nova.composite(data[0], 5, 25);
	nova.composite(data[1], 130, 25);
	nova.print(font, 32, 2, "P1");
	nova.print(font, 155, 2, "P2");
	nova.print(font, 5, 103, p1);
	nova.print(font, 130, 103, p2);
	nova.write(`./images/hg/hgtest.png`);
	await msg.channel.send({files: [`./images/hg/hgtest.png`]});**

**Comment by hipstersmoothie** [comment]
@Allaphy That code wasn't really doing what you wanted to accomplish. The following does what you intended. I also changed the `print` calls to center the text within the picture width so you can put any string there now.

@DavidOndrus  if you provide me an example like this I can try to fix it as well.

```js
const Jimp = require('jimp');

const hguData = [
  {
    user: 'Dan',
    avatar: 'https://avatars0.githubusercontent.com/u/1192452?s=88&v=4'
  },
  {
    user: 'Mary',
    avatar: 'https://avatars3.githubusercontent.com/u/179534?s=460&v=4'
  }
];

const sort1 = 0;
const sort2 = 1;

test();

async function test() {
  const PICTURE_WIDTH = 75;

  const { user: user1, avatar: avatar1 } = hguData[sort1];
  const { user: user2, avatar: avatar2 } = hguData[sort2];

  const images = [avatar1, avatar2];
  const jimps = await Promise.all(images.map(img => Jimp.read(img)));

  jimps.map(img => img.resize(PICTURE_WIDTH, PICTURE_WIDTH));

  console.log('⏳');

  const nova = await Jimp.create(255, 125, '#000000');

  nova.composite(jimps[0], 5, 25);
  nova.composite(jimps[1], 130, 25);

  const font = await Jimp.loadFont(Jimp.FONT_SANS_16_WHITE);

  nova.print(font, 5, 2, { text: 'P1', alignmentX: Jimp.HORIZONTAL_ALIGN_CENTER }, PICTURE_WIDTH);
  nova.print(font, 130, 2, { text: 'P2', alignmentX: Jimp.HORIZONTAL_ALIGN_CENTER }, PICTURE_WIDTH);
  nova.print(font, 5, 103, { text: user1, alignmentX: Jimp.HORIZONTAL_ALIGN_CENTER }, PICTURE_WIDTH);
  nova.print(font, 130, 103, { text: user2, alignmentX: Jimp.HORIZONTAL_ALIGN_CENTER }, PICTURE_WIDTH);

  await nova.writeAsync(`./hgtest.png`);

  console.log('all done!');
  // await msg.channel.send({ files: [`./images/hg/hgtest.png`] });
}
```

**Comment by DavidOndrus** [comment]
My code is really simple in this case - it crashes on the first Jimp line (Jimp.read) - in the case where it happened, I also had sails.log.debug before Jimp.read printing uploadedFiles[0] object a "fd" field was properly there:
```
file.upload(function (err, uploadedFiles) {
      if (err) {
        sails.log.debug(err);
        return res.negotiate(err);
      }

      if (uploadedFiles.length === 0) {
        sails.log.debug('No file was uploaded');
        return res.serverError({error: 'No file was uploaded'})
      }

      const fd = uploadedFiles[0].fd;
      Jimp.read(fd)
        .then(image => {
          const mimeType = image.getMIME();
          return image
            .resize(Jimp.AUTO, 500, Jimp.RESIZE_NEAREST_NEIGHBOR)
            .getBufferAsync(mimeType)
            .then(buffer => {
              //do stuff with buffer
            })
        })
    });
```

**Comment by hipstersmoothie** [comment]
Have you been able to reproduce the error outside of sails? As in a just a simple script no server. I still have now been able to reproduce your issue. Currently running this following with your file with no errors so far:

```js
const Jimp = require('jimp');

test();

async function test() {
  while (true) {
    try {
      const image = await Jimp.read('./exampleImages/636629e3-dcee-4a34-9c07-99f7aadfbd99.jpg');
      await image.resize(Jimp.AUTO, 500, Jimp.RESIZE_NEAREST_NEIGHBOR).writeAsync('test.png');
      console.log('wrote image');
    } catch (error) {
      console.log('CAUGHT ERROR!');
      console.log(error);
    }
  }
}
```

**Comment by DavidOndrus** [comment]
I didn't try outside of sails because Sails is a must for me in the project, it must work everywhere. You said "I still have now been able to reproduce your issue" - so were you ? Or is it a mistake ?

**Comment by hipstersmoothie** [comment]
Oops just a typo. Still *cannot* reproduce. If you cannot reproduce outside of sails then this must be a problem with the interaction between the two. 

I have never used sails so i really need you to put together a repo with as little code as possible that showcases the problem. Otherwise I will not be able to fix this at all.

**Comment by hipstersmoothie** [comment]
To me it seems like this has to be async issues with your code. You've said:

* it happens locally without hitting a network (so it doesn't have to do with jimp's request function)
* it happens when running a sails server with local files.
* it happens sometimes

This leads me to believe that you are not properly awaiting a promise at some point resulting in the image *sometimes* being there. jimp is just using `fs.readFile` to read the file on your local machine. If  `readFile` returns a null buffer that generally means that the file is not yet on your system.

Again without a repo showcasing the issue I cannot debug this issue only theorize.

**Comment by hipstersmoothie** [comment]
@DavidOndrus do you happen to be using webpack?

**Comment by hipstersmoothie** [comment]
I'm gonna close this. If someone can put together an example that errors consistently and share it with me feel free to open an issue!

**Comment by darrensw777** [comment]
Same problem here guys, this fails every time (is it because there is no file extension? If so is there a workaround?):

const image_url = 'https://images.asos-media.com/products/river-island-maxi-dress-with-tie-neck-in-snake-print/10985965-1-snakeprint';

```
Jimp.read(image_url)
                    .then(image => image.write(`/path/to/images/${lastInsertId}.jpg`))
                    .catch(err => {
                        console.error(err);
                    });
```
This is the error I get:

```
{ Error: Could not find MIME for Buffer <null>
18:31:36 0|server   |     at Jimp.parseBitmap (/home/myAccount/node_modules/@jimp/core/dist/utils/image-bitmap.js:108:15)
18:31:36 0|server   |     at Jimp.parseBitmap (/home/myAccount/node_modules/@jimp/core/dist/index.js:498:32)
18:31:36 0|server   |     at /home/myAccount/node_modules/@jimp/core/dist/index.js:440:15
18:31:36 0|server   |     at /home/myAccount/node_modules/@jimp/core/dist/index.js:168:14
18:31:36 0|server   |     at /home/myAccount/node_modules/@jimp/core/dist/request.js:56:9
18:31:36 0|server   |     at IncomingMessage.<anonymous> (/home/myAccount/node_modules/phin/lib/phin.compiled.js:1:2162)
18:31:36 0|server   |     at emitNone (events.js:91:20)
18:31:36 0|server   |     at IncomingMessage.emit (events.js:185:7)
18:31:36 0|server   |     at endReadableNT (_stream_readable.js:974:12)
18:31:36 0|server   |     at _combinedTickCallback (internal/process/next_tick.js:74:11) methodName: 'constructor' }
```

**Comment by hipstersmoothie** [comment]
Write requires a mime type in pretty sure. Nowhere near a computer so u can’t link

**Comment by hipstersmoothie** [comment]
Jimp.auto as the second param

**Comment by hipstersmoothie** [comment]
@darrensw777  The following code works for me.

```js
const Jimp = require("jimp");

test();

async function test() {
  const image = await Jimp.read(
    "https://images.asos-media.com/products/river-island-maxi-dress-with-tie-neck-in-snake-print/10985965-1-snakeprint"
  );

  await image.writeAsync("test.jpg", Jimp.AUTO);
}
```

but your code also just worked when i ran it....

```js
const Jimp = require("jimp");

const image_url =
  "https://images.asos-media.com/products/river-island-maxi-dress-with-tie-neck-in-snake-print/10985965-1-snakeprint";

Jimp.read(image_url)
  .then(image => image.write(`test.jpg`))
  .catch(err => {
    console.error(err);
  });
```

**Comment by hipstersmoothie** [comment]
`write  doesn't return a promise so you should actually use `writeAsync`

```js
const Jimp = require("jimp");

const image_url =
  "https://images.asos-media.com/products/river-island-maxi-dress-with-tie-neck-in-snake-print/10985965-1-snakeprint";

Jimp.read(image_url)
  .then(image => image.writeAsync(`test.jpg`))
  .catch(err => {
    console.error(err);
  });
```

what version are you using?

**Comment by darrensw777** [comment]
Definitely not working still for me with that image url.

Tried:

```
const Jimp = require('jimp');

const image_url =
                    'https://images.asos-media.com/products/river-island-maxi-dress-with-tie-neck-in-snake-print/10985965-1-snakeprint';

Jimp.read(image_url)
    .then(image => image.writeAsync(`test.jpg`))
    .catch(err => {
        console.error(err);
    });
```
Get the following error:
```
0|server   | { Error: Could not find MIME for Buffer <null>
0|server   |     at Jimp.parseBitmap (/home/nw77wish/node_modules/@jimp/core/dist/utils/image-bitmap.js:108:15)
0|server   |     at Jimp.parseBitmap (/home/nw77wish/node_modules/@jimp/core/dist/index.js:498:32)
0|server   |     at /home/nw77wish/node_modules/@jimp/core/dist/index.js:440:15
0|server   |     at /home/nw77wish/node_modules/@jimp/core/dist/index.js:168:14
0|server   |     at /home/nw77wish/node_modules/@jimp/core/dist/request.js:56:9
0|server   |     at IncomingMessage.<anonymous> (/home/nw77wish/node_modules/phin/lib/phin.compiled.js:1:2162)
0|server   |     at emitNone (events.js:91:20)
0|server   |     at IncomingMessage.emit (events.js:185:7)
0|server   |     at endReadableNT (_stream_readable.js:974:12)
0|server   |     at _combinedTickCallback (internal/process/next_tick.js:74:11) methodName: 'constructor' }

```
I only installed this yesterday so presumably the latest version. Installed using NPM

Using with NodeJS v. 11.4.0

OS: Linux 3.10.0-862.14.4.el7.x86_64 x86_64

How can I use and have it not throw a script blocking error? I have a fallback if this fails but it will not fail silently.

**Comment by hipstersmoothie** [comment]
Since i cannot reproduce this locally an example repo is gonna be needed. Also this has nothing to do with the issue we are using to talk about it so another issue would be cool as well.

> How can I use and have it not throw a script blocking error? I have a fallback if this fails but it will not fail silently.

aren't you catching and printing the error? the program should continue right? 

as I said without a reproducible example I will be little help other than just speculation

**Comment by Medosopher** [comment]
let base64_data = "iVBORw0KGgoAAAANSUhEUgAAAO0AAAMiCAYAAACCG0RzAAAAAXNSR0ICQMB9xQAAAAlwSFlzAAAOxAAADsQBlSsOGwAAABl0RVh0U29mdHdhcmUATWljcm9zb2Z0IE9mZmljZX/tNXEAABKsSURBVHja7d0hTJ13FEDxSiQSiURWViKRSCQSWYksColEIpFIJBKJRCIrn6zs7rcsJCPZljTQfQd+JCdZUtH3v/ee7PUk6z6dnZ19Av5P5mdvuMN/cvLnvBwNViDt6XA17OMfOV9mRFqsRdqb4dAs/nVGx6TFmg7yadg1C9KicYzbw8YsSIvOMR4Mt2ZBWnSOcYlQ52ZBWnSOUYQiLWLHKEKRFqFDFKFIi9ghilCkRewQRSjSInaIIhRpETtEEYq0CB2hCEVaxI5QhCItYkcoQpEWsSMUoUiL2BGKUKRF6ABFKNIidoAiFGkRO0ARirSIHaAIRVrEDlCEIi1CxydCkRax4xOhSIvY8YlQpEXs+EQo0iJ2fCIUaRE6PBGKtIgd3uHy9dgsSIvO4X1bMAvSonN4IhRpETu878OOWZAWjaPbWaQ1C9Kic3QiFGkROzoRirSIHZ0IRVrEjk6EIi1CBydCkRaxgxOhSIvYwYlQpEXs4EQo0iJ2cCIUaRE6NhGKtIgdmwhFWsSOTYQiLWLHJkKRFrFjE6FIi9ChiVCkRezQRCjSInZoIhRpETs0EYq0iB2aCEVahI5MhCItYkcmQpEWsSMToUiL2JGJUKRF7MhEKNIidGAiFGkROzARirSIHZgIRVrEDkyEIi1iByZCkRah4xKhSIvYcYlQpEXsuEQo0iJ2XCIUaRE7LhGKtAgdlghFWsQOS4QiLWKHJUKRFrHDEqFIi9hhiVCkReioRCjSInZUIhRpETsqEYq0iB2VCEVaxI5KhCItQgclQpEWsYMSoUiL2EGJUKRF7KBEKNIidlAiFGkROiYRirSIHZMIRVrEjkmEIi1ixyRCkRaxYxKhSIvQIe0OT2ZBWnQOSYQiLWKHdD6cmgVp0Tmk2+HALEiLziFthm2zIC0aRyRCkRaxIxKhSIvYEYlQpEXsiEQo0iJ2RCIUaRE6IBGKtIgdkAj1+2d+MlySFr96QCLU75/5838CaSD4lQMSoUiL2AGJUKRF6HhEKNIidjwiFGkROx4RirSIHY8IRVrEjkeEIi1ChyNCkRaxwxGhSIvY4YhQpEXscEQo0iJ2OCIUaRE6GhGKtIgdjQhFWsSORoQiLWJHI0KRFrGjEaFIi9DBiFCkRexgRCjSInYwIhRpETsYEYq0iB2MCEVahI5FhCItYsciQpEWsWMRoUiL2LGIUKRF7FhEKNIidCgiFGkROxQRirSIHYoIRVrEDkWEIi1ihyJCkRahIxGhSIvYkYhQpEXsSEQo0iJ2JCIUaRE7EhGKtAgdiAhFWsQORIQiLWIHIkKRFrEDEaFIi9iBiFCkReg4RCjSInYcIhRpETsOEYq0iB2HCEVaxI5DhCItQoexNzyaBWnROYyj4dosSIvOYVwMX82CtOgcxt2wbxakRecwRCjSInQUIhRpETsKEYq0iB2FCEVaxI5ChCItYkchQpEWoYMQoUiL2EGIUKRF7CBEKNIidhAiFGkROwgRirQIHYMIRVrEjkGEIi1ixyBCkRaxYxChSIvYMYhQpEXoEEQo0iJ2CCIUaRE7BBGKtIgdgghFWsQOQYQiLUJHIEKRFrEjEKFIi9gRiFCkRewIRCjSInYEIhRpEToAEYq0iB2ACEVaxA5AhCItYgcgQpEWsQMQoUiL0PJFKNIitnwRirSILV+EIi1iyxehSIvY8kUo0iK0eBGKtIgtXoQiLWKLF6FIi9jiRSjSIrZ4EYq0CC1dhCItYksXoUiL2NJFKNIitnQRirSILV2EIi1CCxehSIvYwkWo/g6fm4SBfLCFI7vDq+GYtB9n4SIUaRFbuAhFWoSWLUKRFrFli1CkRWzZIhRpEVu2CEVaxJb9Y9gyC9KisejPw4NZkBadRR8vyzYL0qKzaBGKtIgt+n74YhakRWfRIhRpEVqyCEVaxJYsQpEWsSWLUKRFbMkiFGkRW7IIRVqEFixCkRaxBYtQpEVswSIUaRFbsAhFWsQWLEKRFqHlilCkRWy5IhRpEVuuCEVaxJYrQpEWseWKUKRFaLEiFGkRW6wIRVrEFitCkRaxxYpQpEVssSIUaRFaqghFWsSWKkKRFrGlilCkRWypIhRpEVuqCEVahBYqQpEWsYWKUKRFbKEiFGkRW6gIRVrEFipCkRahZYpQpEVsmSIUaRFbpghFWsSWKUKRFrFlilCkRWiRIhRpEVukCEVaxBYpQpEWsUWKUKRFbJEiFGkRWqIIRVrElihCkRaxJYpQpEVsiSIUaRFboghFWoQWKEKRFrEFilCkRWyBIhRpEVugCEVaxBYoQpEWoeWJUKQ1kNjyRCjSGkhseSIUaQ0ktryH5SuyWZAWjcVtDRsRirToLE6EIi1pY4s7GS7NgrToLO5yEdcsSIvO4kQo0pI2tDQRirSkjS1NhCItaWNLE6FIS9rY0kQo0pI2tjQRirSkDS1MhHIDpI0tTIRyA6SNLUyEcgOkjS1MhHIDpI0tTIRyA6QNLUuEAmljyxKhQNrYskQokDa2LBEKpI0tS4QCaUOLEqFA2tiiRCiQNrYoEQqkjS1KhAJpY4sSoUDa0JJEKJA2tiQRCqSNLUmEAmljSxKhQNrYkkQokDa0IBEKpI0tSIQCaWMLEqHw8iZuhkPSrndBIhRe3sTdsE/a9S5IhAJpQ8sRoUDa2HJEKJA2thwRCqSNLUeEAmljyxGhQNrQYkQokDa2GBEKpI0tRoQCaWOLEaFA2thiRCiQNrQUEQqkjS1FhAJpY0sRoUDa2FJEKJA2thQRCqQNLUSEAmljCxGhQNrYQkQokDa2EBEKpI0tRIQCaUPLEKFA2tgyRCiQNrYMEQqkjS3j+X/3AJC2sQwRCqQNLWKJUD/MAqTtLOLLcG8WIG1nEV+HC7MAaTuLEKFA2tgiRCiQNrQEEQqkjS1BhAJpY0sQoUDa2BJEKJA2tgQRCqQNLUCEAmljCxChQNrYAkQokDa2ABEKpI0tQIQCaUPDF6FA2tjwRSiQNjZ8EQqkjQ1fhAJpY8MXoUDa0OBFKJA2NngRCqSNDV6EAmljgxehQNrY4EUokDY0dBEKpI0NXYQCaWNDF6FA2tjQRSiQNjZ0EQqkDQ1chAJpYwMXoUDa2MBFKJA2NnARCqSNDVyEAmlDwxahQNrYsEUokDY2bBEKpI0NW4QCaWPDFqFA2tCgRSiQNjZoEQqkjQ1ahAJpY4MWoUDa2KBFKJA2NGQRCqSNDVmEAmljQxahQNrYkEUokDY2ZBEKpA0NWIQCaWMDFqFA2tiARSiQNjZgEQqkjQ1YhAJpQ8MVoUDa2HBFKJA2NlwRCqSNDfd6ODILkLYz3MdhzyxA2sZgt4eNWYC0ncHuL8M1C5C2M1gRCqSNDVaEAmljgxWhQNrQUEUokDY2VBEKpI0NVYQCaWNDFaFA2thQRSiQNjRQEQqkjQ1UhAJpYwMVofDWN/b8t6EYyOsMVITCW9/Y07BL2tcbqAgF0oaGKUKBtLFhilAgbWyYIhRIGxumCAXSxoYpQoG0oUGKUCBtbJAiFEgbG6QIBdLGBilCgbSxQYpQIG1oiCIUSBsboggF0saGKEKBtLEhilAgbWyIIhRIGxqgCAXSxgYoQoG0sQGKUCBtbIAiFEgbG6AIBdKGhidCgbSx4YlQIG1seCIUSBsbnggF0saGJ0KBtKHBiVAgbWxwIhRIGxucCAXSxgYnQoG0scGJUCBtaGgiFEgbG5oIBdLGhiZCgbSxoYlQIG1saCIUSBsamAgF0sYGJkKBtLGBiVAgbWxgIhRIGxuYCAXShoYlQoG0sWGJUCBtbFinw7lZgLSdYd0Mh2YB0gaHBZB2/YMSoUDa2KAOhluzAGk7gxKhQNrYoEQokLY6KIC06x+SCAXSxoYkQoG0sSGJUCBtbEgiFEhbHRJA2vUPSIQCaWMDEqFA2tiARCiQNjYgEQqkrQ4IIO36hyNCgbSx4YhQIG1sOCIUSBsbjggF0laHA5B2/YMRoUDa2GBEKJA2NhgRCqSNDUaEAmmrgwFIu/6hiFAgbWwoIhRIGxuKCAXSxoYiQoG01aEApF3/QEQokDY2EBEKpI0NRIQCaWMDEaFA2upAANKufxgiFEgbG4YIBdLGhiFCgbSxYYhQIG11GABp1z8IEQqkjQ1ChAJpY4MQoUDa2CBEKJC2OgiAtOsfgggF0saGIEKBtLEhiFAgbWwIIhRIWx0CQNr1D0CEAmljAxChQNrYAEQo1G52s3xD/MjSilCo3ezP53/+6F81ANKu//EiFEgbe/zh8vXYIYC0ncd/W3AIIG3n8SIUSBt7/PdhxyGAtI2H7yzSOgKQtvNwEQqkjT1chAJpYw8XoUDa2MNFKJA29GgRCqSNPVqEAmljjxahQNrYo0UokDb2aBEKpA09WIQCaWMPFqFA2tiDRSiQNvZgEQqkjT1YhAJpQ48VoUDa2GNFKJA29lgRCqSNPVaEAmljjxWhQNrQQ0UokDb2UBEKpI09VIQCaWMPFaFA2thDRSiQNvRIEQqkjT1ShAJpY48UoUDa2CNFKJA29kgRCqQNPVCEAmljDxShQNrYA0UokDb2QBEKpI09UIQCaUOPE6FA2tjjRCiQNvY4EQqkjT1OhAJpY48ToUDa0MNEKJA29jARCqSNPUyEAmljDxOhQNrYw0QokDb0KBEKpI09SoQCaWOPEqFA2tijRCiQNvYoEQqkDT1IhAJpYw8SoUDa2INEKJA29iARCqSNPUiEAmlDj9kdniwWpO08RoQCaWOPOR9OLRak7TzmdjiwWJC285jNsG2xIG3jISIUSBt7iAgF0sYeIkKBtLGHiFAgbewhIhRIG3qECAXSxh4hQoG0sUeIUCBt7BEiFEgbe4QIBdKGHiBCgbSxB4hQIG3sASIUSBt7gAgF0sYeIEKBtKEPL0KBtLEPL0KBtLEPL0KBtLEPL0KBtLEPL0KBtKEPLkKBtLEPLkKBtLEPLkKBtLEPLkKBtLEPLkKBtKEPLUKBtDFpRSiQNiatCAXSxqQVoUDamLQiFEhbkVaEAml70opQIG1MWhEKH1nY7eWPhzVpRSh8ZGn/9sfDyocWoUDairQiFEjbk1aEAmlj0opQIG1MWhEKpI1JK0KBtBVpRSigJ60IBdLGpBWhQNqYtCIUSBuTVoQCaSvSilBAT1oRCohJK0IBMWlFKCAmrQgFVKQVoYCetCIUEJNWhAJi0opQQExaEQqoSDs/e8OjZQEdaY+Ga8sCOtJeDF8tC+hIezfsWxbQkVaEAirSilBAT1oRCohJK0IBMWlFKCAmrQgFVKQVoYCetCIUEJNWhAJi0opQQExaEQqoSCtCAT1pRSggJq0IBcSkFaGAmLQiFFCRVoQCetKKUEBMWhEKiEkrQgExaUUooCKtCAX0pBWhgJi0IhQQk1aEAmLSilBARVoRCuhJK0IBMWlFKCAmrQgFxKQVoYCKtCIU0JNWhAJi0opQQExaEQqISStCARVpRSigJ60IBcSkFaGAmLQiFBCTVoQCKtKKUEBPWhEKiEkrQgExaUUoICatCAVUpBWhgJ60IhQQk1aEAmLSilBATFoRCqhIK0IBPWlFKCAmrQgFxKQVoYCYtD+GLQsAAtLOz+fhwfCBjrTHw5XhAx1pRSggJu398MXwgY60IhRQkVaEAnrSilBATFoRCohJK0IBMWlFKKAirQgF9KQVoYCYtCIUEJNWhAJi0opQQEVaEQroSStCATFpRSjgjb65vtVvIkIBr+fT/vL3rL21tCIUUJFWhAJ60opQQExaEQqISStCATFpRSigIq0IBfSkFaGAmLQiFBCTVoQCYtKKUEBFWhEK6EkrQgExaUUoICatCAXEpBWhgIq0IhTQk1aEAmLSilBATFoRCohJK0IBFWlFKKAnrQgFxKQVoYCYtCIUEJNWhAIq0opQQE9aEQqISStCATFpRSggJq0IBVSkFaGAnrQiFBCTVoQCYtKKUEBMWhEKqEgrQgE9aUUoICatCAXEpH1YviIbKBCQdolPw0aEAjrSilBATNqT4dIwgY60l4u4hgl0pBWhgIq0IhTQk1aEAmLSilBATFoRCohJK0IBFWlFKKAnrQgFxKQVoYCYtCIUEJNWhAIq0opQQE9aEQqISStCATFpRSggJq0IBVSkFaGAnrQiFBCTVoQCYtKKUEBMWhEKqEgrQgE9aUUoICatCAXEpBWhgJi0IhRQkVaEAnrSilBATFoRCohJK0IBMWlFKKAirQgF9KQVoYCYtCIUEJNWhAJi0opQQEVaEQroSStCATFpRSggJq0IBcSkFaGAirQiFNCTVoQCYtKKUEBMWhEKiEkrQgEVaUUooCetCAXEpBWhgJi0IhQQk1aEAirSilBAT1oRCohJK0IBMWmvhqflF/Bu+Il3xelLaXf/shnvBP+Ger/8AamUt6f/h5yRAAAAAElFTkSuQmCC";
	let buffer = Buffer.from(base64_data, 'base64').buffer;
	let image = await (Jimp.read(buffer));
	base64_data = "iVBORw0KGgoAAAANSUhEUgAAAdQAAADlCAYAAADqdgbQAAAAAXNSR0ICQMB9xQAAAAlwSFlzAAAOxAAADsQBlSsOGwAAABl0RVh0U29mdHdhcmUATWljcm9zb2Z0IE9mZmljZX/tNXEAAAl0SURBVHja7d0vUBtbH8fhKysrkZWVyMpKJBKJRFZWgkIiKysrkZWVlZVVnbqQXbHMSUREJtlLZnjfS3YD5ED+nLP7iMdwb0tmM51Pf5fvhX8uLi7+AQBex0MAAEEFgN2bDcPZrBxfzQbV6bSsDgUVAF5gXoyuZ0Wo/29YHXswABB7oRbh5mFQ60H1zoMBgAh1NTlYuk6LcLv4uIcDADHX6bA6fhjUeRG+CyoAxAa1COdLF2o5uhRUAIi0apAkqAAQf6G2BkmCCgARHhskCSoAxFynjwySBBUAYoL6yCBJUAEgwuIiXTVIElQAiLtQb1cNkgQVANa0iGdjkHTz8J97SACwznXaGiSNrgUVAGKDWo4uGxfquaACQKSnBkmCCgDrXqjNQVI1ORBUAIjw3CBJUAFgnev0mUGSoALAOkEtx1dPDZIEFQDWMC9HP54aJAkqAKxzoT4zSBJUAHhGPRi/f26QJKgA8Nx1Ogwnzw2SBBUAngtqc5A0DJ8FFQAiNQdJ0zIcCSoAxF6orUFS9VZQASDCikHS38f+XQ8MAB67TtccJAkqADwV1DUHSYIKAE+YF+HnOoMkQQWApy7UIkzWGSQJKgA8YlpWh+sOkgQVAB67TgfV6bqDJEEFgMeC2hokVZ8EFQAitQZJg+qjoAJA7IUaMUgSVABYoTVIKsPv536NBwcAzeu0MUi6C+o3QQWA2KBGDpIEFQBWmBfhV8wgSVABoKH+U7+JHSQJKgA0vGSQJKgA0DAbhrPYQZKgAkAzqEX4EjtIElQAaHjJIElQAeCBlYOku48JKgBEaA6SFtfqur/WAwSAe+1B0uiroAJAbFBfOEgSVAB4oDVIKscfBBUAItwPkuqXDJIEFQDuLa7Rlw6SBBUA7i2+XvrSQZKgAsD/gnoX0JcOkgQVAO69ZpAkqABw8fpBkqACwMXrB0mCCgAXKwZJRfgiqAAQG9TWICmcCSoARAc1/F4eJFWHggoAEeqqetv4z72T2EGSoALQe4sfIP7aQZKgAtB7mxgkCSoAglqGb68dJAkqAIK6gUGSoALQa5saJAkqAL22qUGSoALQa5saJAkqAP0OanOQNKhOBRUAYoNahL+bGCQJKgC9tWqQ9Jrfz0MFoJemZThqDJJ+CioARJoNw+elC7UcXwkqAESaF6PrTQ2SBBWA/l6oGxwkCSoAvbTpQZKgAtBLmx4kCSoAvbTpQZKgAtBLmx4kCSoA/bxQG4OkejB+L6gAEKGuJgeNQdLtJn5fDxeAfl2nw+p4aZBUjn4IKgDEBrUI55seJAkqAL3TGiQNw4mgAkD8hXqz6UGSoALQK9saJAkqAP26Trc0SBJUAPoV1C0NkgQVgF7Z1iBJUAHo24V6u41BkqAC0Bv1oHq3rUGSoALQn+u0OUgqwndBBYDYoJajy+VB0uhSUAEg0uIiXR4kVceCCgCxF2prkFS9E1QAiLDtQZKgAtCP63TLgyRBBaAfQd3yIElQAeiFbQ+SBBWAflyoWx4kCSoAnbf49oLbHiQJKgDdv06H4WTbgyRBBaD7QS3HV40L9VxQASDS4oeIb3uQJKgAdP9CbQ6SqsmBoAJAhBWDpJttfS4PHIDuXqetQdLoWlABIDaoOxokCSoAnbarQZKgAtDtC3VHgyRBBaCzpmV1uKtBkqAC0N3rdFCd7mqQJKgAdDeoOxwkCSoAnTUvws9dDZIEFYDuXqhFmCwPkqq3ggoAEVYMkv5u+3N68AB07zrd8SBJUAHoZlCbg6Rh+CyoABCpOUialuFIUAEg9kLd8SBJUAHonH0MkgQVgO5dp8NwtutBkqAC0L2gFuHLrgdJggpA58yL8GvXgyRBBaBT6j/1m30MkgQVgE5pDZLK8HtXn9sbAEBnNAdJd0H9JqgAEBvU1iCp+iSoABCpNUgaVB8FFQAi7HOQJKgAdMa0HH/Y1yBJUAHojMXXS/c1SBJUALoT1HL0dV+DJEEFoDP2OUgSVAA64X6QVO9rkCSoAHTCvgdJggpAJ7QHSaOvggoAsUHd8yBJUAHohPaPbBt/EFQAiLBykHT3MUEFgAiL/z3mYUwX1+o+Xoc3A4CspTBIElQA8g9qGb7te5AkqAB0Iai/9z1IElQAsrb4bkgpDJIEFYCspTJIElQAspbKIElQAcg7qIkMkgQVgNyD2hgkVYeCCgARVgySJvsaJAkqANlKaZAkqABkazYMnxsX6hdBBYBI82J0vTxICmeCCgCxF2oR/qYySBJUALKU2iBJUAHI0rQMRykNkgQVgCylNkgSVACylNogSVAByPNCTWyQJKgAZCfFQZKgApDfdTqsjhuDpJ8pvC5vDgB5BbUI58s/sm18JagAEKk1SBpUp4IKAPEX6k1qgyRBBSArdTU5aA6SUnlt3iAA8rlOEx0kCSoAeQU10UGSoAKQlVQHSYIKQG4XapKDJEEFIBspD5IEFYB8rtPmIKkc/RBUAIgNajm6THWQJKgAZGNehO+NH9l2IqgAEHuhFuH2YVDrwfi9oAJAhHpQvWsMkm5Te43eKADSv04THyQJKgB5BDXxQZKgApCF1AdJggpAHhdq4oMkQQUgeTkMkgQVgPSv0wwGSYIKQPpBLcdXqQ+SBBWA5C0u0uVBUnUsqAAQe6G2BknVO0EFgAiLNW8OgyRBBSDt63QYTpYGSUX4LqgAEBvU1iBpdCmoABApl0GSoAKQ9oWaySBJUAFIVk6DJEEFIN3rNKNBkqACkG5QMxokCSoAybq7SH/mMkgSVADSvVCLMMllkCSoACRpWlaHjUHSTeqv2RsHQHrX6aA6XR4kja4FFQBig9ocJBXhXFABIFJugyRBBSDNC7U5SKomB4IKABFyHCQJKgDpXacZDpIEFYD0glqEL7kNkgQVgOTMi/Art0GSoAKQlPpP/SbHQZKgApCUXAdJggpAUmbDcJbjIElQAUgrqM1B0jB8FlQAiNQcJE3LcCSoABBh9SCpeiuoABBhxSDpb06v35sIQBJyHiQJKgDpBLUcfc11kCSoACQj50GSoAKQhPtBUp3rIElQAUjCtBx/yHmQJKgAJGE2rD7lPEgSVADSCGprkFR9ElQAiNQaJA2qj4IKABG6MEgSVIAOWlx3OWl+Q4dZGX7n+Nz/ewPKcDQvRz+gy+7+sN42/yYMJKYM37IO6mxQnXojAdi7DAdJggpAaib1YPxeUAHYu2y/LFOE77NhOMn1a9f/rayqyUFuX8iGWDkuB4E8/AvRP8zXwAKJzQAAAABJRU5ErkJggg==";
	buffer = Buffer.from(base64_data, 'base64').buffer;
	image = await (Jimp.read(buffer));


This is producing the error. The second image on its own doesn't. It's as if the Jimp library is in some way affected by reading the first image.
This may go far deeper than I thought.

**Comment by RMCoder198** [comment]
I solved this issue by using promise method  as suggested in docs instead of using callback method. 
Another reason i am getting this error is because of having too many images on destination directory.
@hipstersmoothie  you might try to reproduce it by having 50+ images inside destination directory.

**Comment by mayank1990dec** [comment]
I am also facing same issue, when I gave image path, like following 
 **let image = await Jimp.read("./assets/not-button/1.jpg");**

But when import my image and then pass it to JIMP.read(), it works, not sure what the issue is 

**import notButton1 from './assets/not-button/1.jpg';**
 **let image = await Jimp.read("notButton1");**

**Comment by FridayJew** [comment]
The same error, the files are received from the Internet and saved on the disk, but are not read by the jimp module. There are no visible problems with them.

**Comment by codan84** [comment]
I get this error without fail every time I read certain images from a URL.  

My app does a bing image search and then tries to read a random image. some of them read fine, some of them throw an error as described in this issue.  

The ones that throw an error, throw it always.  

One of the offending images: https://s-media-cache-ak0.pinimg.com/736x/c9/8f/e1/c98fe17dc7de72bb29c34a0c79ef5762.jpg

Here is a [repo](https://github.com/codan84/jimp-bug) that reproduces the issue (I am running this on node `v10.16.2`).  

I can provide many more image URLs that always fail.

**Comment by codan84** [comment]
> Since i cannot reproduce this locally an example repo is gonna be needed. Also this has nothing to do with the issue we are using to talk about it so another issue would be cool as well.
> 
> > How can I use and have it not throw a script blocking error? I have a fallback if this fails but it will not fail silently.
> 
> aren't you catching and printing the error? the program should continue right?
> 
> as I said without a reproducible example I will be little help other than just speculation

@hipstersmoothie I have provided an example repo to reproduce the issue above ⬆️

**Comment by FridayJew** [comment]
> > Since i cannot reproduce this locally an example repo is gonna be needed. Also this has nothing to do with the issue we are using to talk about it so another issue would be cool as well.
> > > How can I use and have it not throw a script blocking error? I have a fallback if this fails but it will not fail silently.
> > 
> > 
> > aren't you catching and printing the error? the program should continue right?
> > as I said without a reproducible example I will be little help other than just speculation
> 
> @hipstersmoothie I have provided an example repo to reproduce the issue above ⬆️

I think no one will answer. Topic is closed.

**Comment by codan84** [comment]
> > > Since i cannot reproduce this locally an example repo is gonna be needed. Also this has nothing to do with the issue we are using to talk about it so another issue would be cool as well.
> > > > How can I use and have it not throw a script blocking error? I have a fallback if this fails but it will not fail silently.
> > > 
> > > 
> > > aren't you catching and printing the error? the program should continue right?
> > > as I said without a reproducible example I will be little help other than just speculation
> > 
> > 
> > @hipstersmoothie I have provided an example repo to reproduce the issue above ⬆️
> 
> I think no one will answer. Topic is closed.

Submitted a new, open one ;)

**Comment by Priyanshu348** [comment]
const Discord = require('discord.js');
const db = require("quick.db")
var jimp = require('jimp');

module.exports = async (client, member) =>{
	let wChan = db.fetch(`${member.guild.id}`)
	
	if(wChan == null) return;
	
	if(!wChan) return;
	
let font = await jimp.loadFont(jimp.FONT_SANS_32_BLACK) //We declare a 32px font
  let font64 = await jimp.loadFont(jimp.FONT_SANS_64_WHITE) //We declare a 64px font
  let bfont64 = await jimp.loadFont(jimp.FONT_SANS_64_BLACK)
  let mask = await jimp.read('https://i.imgur.com/552kzaW.png') //We load a mask for the avatar, so we can make it a circle instead of a shape
  let welcome = await jimp.read('http://rovettidesign.com/wp-content/uploads/2011/07/clouds2.jpg') //We load the base image

  jimp.read(member.user.displayAvatarURL).then(avatar => { //We take the user's avatar
    avatar.resize(200, 200) //Resize it
    mask.resize(200, 200) //Resize the mask
    avatar.mask(mask) //Make the avatar circle
    welcome.resize(1000, 300)
	
  welcome.print(font64, 265, 55, `Welcome ${member.user.username}`) //We print the new user's name with the 64px font
  welcome.print(bfont64, 265, 125, `To ${member.guild.name}`)
  welcome.print(font64, 265, 195, `There are now ${member.guild.memberCount} users`)
  welcome.composite(avatar, 40, 55).write('Welcome2.png') //Put the avatar on the image and create the Welcome2.png bot
  try{
  member.guild.channels.get(wChan).send(``, { files: ["Welcome2.png"] }) //Send the image to the channel
  }catch(e){
	  // dont do anything if error occurs
	  // if this occurs bot probably can't send images or messages
  }
  })

	
	

}
 ○This is a event guildmemberaddup up○
Pls help me and tell me how to fix

**Comment by tomisinteazer** [comment]
this issue for me was caused by the .DS_store i removed it from the array and all worked fine

**Comment by seiyria** [comment]
@tomisinteazer thanks, this helped me. Erroneously assuming fs.readDirSync would only return the png files I was expecting was a mistake; I had to filter the array but files that had png in the name.
